providers/frame/common.ts

```diff
@@ -14,11 +14,21 @@ export async function createCustomChainSpec(
   })
   const chainSpec = JSON.parse(new TextDecoder().decode((await buildSpecCmd.output()).stdout))
   const balances: [string, number][] = chainSpec.genesis.runtime.balances.balances
-  for (let i = 0; i < DEFAULT_TEST_USER_COUNT; i++) {
-    balances.push([
-      ss58.encode(networkPrefix, testUser(i).publicKey),
-      DEFAULT_TEST_USER_INITIAL_FUNDS,
-    ])
+  try {
+    balances.push(
+      ...JSON.parse(
+        await Deno.readTextFile(
+          new URL(`./test_users/network_${networkPrefix}_user_balances.json`, import.meta.url),
+        ),
```

Comment:
Each `network_<prefix>_user_balances.json` weights ~7mb.
I'm not sure what would be best
- keep them in Git
- upload them the something like S3 and cache them in a CDN

WDYT?

Reply:
Hmm, I didn't think about how large the files would be... Can you test to see how much time each part of the generation takes? (text encoding / hashing / sr25519 from seed / ss58 generation) If the text encoding and/or hashing are expensive, we could adopt a different approach for the seed generation, but if the majority of the time is the sr25519 or ss58, we might be out of luck.

Reply:
Ah, ok. In that case, instead of pre-computing the full addresses, could you just pre-compute the public keys, and then save them into a file using `$.array($.sizedUint8Array(32))`? That way it can be the same file across chains, and it should be only around 3.2 MB.

Reply:
And if the file is only around 3.2 megs, you can put that in git for now, like we do with the dprint formatter. Later, once I take a look at #709, that file can potentially be moved from git into an s3 cache.